### PR TITLE
feat: Add comment to MongoDB query via `Parse.Query.comment`

### PR DESCRIPTION
**Pull request by Meglali20**
SDK changes for : https://github.com/parse-community/parse-server/issues/8799

### CODE REVIEW

**Comment by mtrezza** [comment]
CI fails with:

> 1) Parse Query can query and send a comment
>   - Error: Invalid parameter for query: comment

I believe the tests are not running with the latest alpha version of Parse Server because the package-lock file usually specifies a commit hash so it's locked to a specific version. Running `npm i -DE parse-server@7.0.0-alpha.23` should fix that. However, there's https://github.com/parse-community/Parse-SDK-JS/issues/1979 which we should fix in the coming days, so some tests may fail that are unrelated to this PR.

**Comment by Meglali20** [comment]
Yes, I was expecting this error. for me, all tests passed when I used parse server as local package after making changes locally related to https://github.com/parse-community/parse-server/pull/8928

**Comment by mtrezza** [comment]
Could you upgrade Parse Server like I've mentioned?

**Comment by mtrezza** [comment]
There has been some issue with the package-lock file; I've reverted it and ran `npm i -DE parse-server@7.0.0-alpha.25`.

**Comment by mtrezza** [comment]
I believe we need to upgrade the Parse JS SDK to support only Node 18 and 20, the same node versions as Parse Server 7. In a separate PR, before we can merge this PR. https://github.com/parse-community/Parse-SDK-JS/pull/2063

**Comment by mtrezza** [comment]
Tests are failing because the Parse JS SDK test suite needs to be updated for Parse Server 7. For example, allowClientClassCreation needs to be set to true. https://github.com/parse-community/Parse-SDK-JS/issues/1979

**Comment by Meglali20** [comment]
When running tests locally most of them pass, did you figure out the issue for this?

**Comment by mtrezza** [comment]
Did you download the latest commit of this branch? I would be surprised if it passed locally. They likely pass when testing with an older version of Parse Server, but not with the latest alpha version as in this branch.

**Comment by Meglali20** [comment]
You're correct tests are failing, didn't notice you changed to alpha 25 I had to run `npm i -DE parse-server@7.0.0-alpha.25` after that tests started failing.

**Comment by mtrezza** [comment]
Yes, I recommend to pull the latest commit in this PR to your local clone, so that you use the same code, delete your current `node_modules` directory, then run just `npm i`. Otherwise you may see different errors.

**Comment by mtrezza** [comment]
Tests are passing now; could you check the codecov? It seems that the new tests are not testing all lines of the new feature. After that it should be good to merge.

File: src/ParseQuery.js [hunk 1]
```diff
@@ -2111,6 +2122,26 @@ class ParseQuery {
...
+
+  /**
+   * Sets a comment to the query so that the query
+   *can be identified when using a the profiler for MongoDB.
+   *
+   * @param {string} value a comment can make your profile data easier to interpret and trace.
+   * @returns {Parse.Query} Returns the query, so you can chain this call.
+   */
+  comment(value: string): ParseQuery {
+    if (typeof value === 'undefined') {
```
Comment (mtrezza) [inline src/ParseQuery.js]:
```suggestion
    if (value == null) {
```

Simplify, so that undefined and null values are considered "no value"?

File: src/ParseQuery.js [hunk 2]
```diff
@@ -2111,6 +2122,26 @@ class ParseQuery {
...
+      delete this._comment;
+      this._comment = value;
```
Comment (mtrezza) [inline src/ParseQuery.js]: Why is this needed after delete?

```suggestion
```

**Review by mtrezza — APPROVED** [review]
Looks good! Codecov is fixed and all tests pass, thanks for the PR!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-alpha.3](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.0.0-alpha.3)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.0.0)
